src/m_Do/m_Do_ext.cpp

```diff
@@ -992,43 +1015,333 @@ s32 mDoExt_resIDToIndex(JKRArchive* p_archive, u16 id) {
+            info1.mTranslate.z = info1.mTranslate.z * f31 + oldTransInfo->mTranslate.z * oldFrameRate;
+            info1.mScale.x = info1.mScale.x * f31 + oldTransInfo->mScale.x * oldFrameRate;
+            info1.mScale.y = info1.mScale.y * f31 + oldTransInfo->mScale.y * oldFrameRate;
+            info1.mScale.z = info1.mScale.z * f31 + oldTransInfo->mScale.z * oldFrameRate;
+        }
+    } else if (param_0 == model->getModelData()->getJointNum() - 1) {
+        mOldFrame->onOldFrameFlg();
+    }
+    Mtx mtx;
+    if (mBeforeCallback) {
+        mBeforeCallback(field_0x58, param_0, &info1, &quat3);
+    }
+    mDoMtx_quat(mtx, &quat3);
+    mDoExt_setJ3DData(mtx, &info1, param_0);
+    if (mAfterCallback) {
+        mAfterCallback(field_0x58, param_0, &info1, &quat3);
+    }
+    *oldQuat = quat3;
+    *oldTransInfo = info1;
+    if (param_0 == model->getModelData()->getJointNum() - 1) {
+        mOldFrame->decOldFrameMorfCounter();
+    }
 }
 
 /* 80012548-800125CC       .text initOldFrameMorf__22mDoExt_MtxCalcOldFrameFfUsUs */
-void mDoExt_MtxCalcOldFrame::initOldFrameMorf(f32, u16, u16) {
-    /* Nonmatching */
+void mDoExt_MtxCalcOldFrame::initOldFrameMorf(f32 param_0, u16 param_1, u16 param_2) {
+    if (param_0 > 0.0f) {
+        mOldFrameMorfCounter = param_0;
+        field_0x8 = 1.0f / param_0;
+        mOldFrameRate = 1.0f;
+        field_0x10 = 1.0f;
+        field_0x14 = 1.0f;
+        decOldFrameMorfCounter();
+    } else {
+        mOldFrameMorfCounter = 0.0f;
+        field_0x8 = 0.0f;
+        mOldFrameRate = 0.0f;
+        field_0x10 = 0.0f;
+        field_0x14 = 0.0f;
+    }
+    mOldFrameStartJoint = param_1;
+    mOldFrameEndJoint = param_2;
 }
 
 /* 800125CC-80012650       .text decOldFrameMorfCounter__22mDoExt_MtxCalcOldFrameFv */
 void mDoExt_MtxCalcOldFrame::decOldFrameMorfCounter() {
-    /* Nonmatching */
+    if (!(mOldFrameMorfCounter > 0.0f)) {
+        return;
+    }
+    mOldFrameMorfCounter -= 1.0f;
+    if (mOldFrameMorfCounter <= 0.0f) {
+        mOldFrameMorfCounter = 0.0f;
+        field_0x8 = 0.0f;
+        mOldFrameRate = 0.0f;
+    }
+    field_0x14 = field_0x10;
+    field_0x10 = mOldFrameMorfCounter * field_0x8;
+    if (field_0x14 > 0.0f) {
+        mOldFrameRate = 1.0f - (field_0x14 - field_0x10) / field_0x14;
+        return;
+    }
+    mOldFrameRate = 0.0f;
 }
 
 /* 80012650-800129E4       .text __ct__14mDoExt_McaMorfFP12J3DModelDataP25mDoExt_McaMorfCallBack1_cP25mDoExt_McaMorfCallBack2_cP15J3DAnmTransformifiiiPvUlUl */
-mDoExt_McaMorf::mDoExt_McaMorf(J3DModelData*, mDoExt_McaMorfCallBack1_c*, mDoExt_McaMorfCallBack2_c*, J3DAnmTransform*, int, f32, int, int, int, void*, u32, u32) {
-    /* Nonmatching */
+mDoExt_McaMorf::mDoExt_McaMorf(J3DModelData* modelData, mDoExt_McaMorfCallBack1_c* callback1, mDoExt_McaMorfCallBack2_c* callback2, J3DAnmTransform* anmTransform, int param_4, f32 param_5, int param_6, int param_7, int param_8, void* param_9, u32 param_10, u32 param_11) {
```

Comment:
`J3DModelData* modelData, mDoExt_McaMorfCallBack1_c* callback1, mDoExt_McaMorfCallBack2_c* callback2, J3DAnmTransform* anmTransform, int loopMode, int speed, int startFrame, int duration, int hasSound, void* soundRes, u32 modelFlag, u32 differedDlistFlag` is what I have locally.